lib/inspec/resources/service.rb

```diff
@@ -154,7 +154,7 @@ def select_service_mgmt # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCom
         SysV.new(inspec, service_ctl)
       when "mac_os_x"
         LaunchCtl.new(inspec, service_ctl)
-      when "windows"
+      when /^windows/
```

Comment:
I wonder if there are other cases that this does not match.
For example a string starting `Microsoft` or a string with uppercase `Windows`.
Note that in train, the name for a Windows OS may be the result of a PowerShell command that we can't predict the exact result for.

For this resource, it may be safer to have this method rely on `os.windows?`

For example, you could put this line at line 106:

`return WindowsSrv.new(inspec) if os.windows?`

...and then delete this `when /^windows/` case.

Reply:
You're absolutely right. This prevents some Windows cases that may not match. I check it on my local lab and if it works I update the MR.